synapse/storage/_base.py

```diff
@@ -112,7 +112,10 @@ def _attempt_to_invalidate_cache(
         if key is None:
             cache.invalidate_all()
         else:
-            cache.invalidate(tuple(key))
+            # Prefer any local-only invalidation method. Invalidating any non-local
+            # cache beforehand is the responsibility of the worker making the change.
```

Comment:
This function is *not* only used when we receive an update over replication: it can also be used on the worker making the change itself. So, I think this comment is misleading (even if the code change itself is valid).

Reply:
https://github.com/matrix-org/synapse/pull/13308/commits/e90f402031d3960b4c1bc9434d402912df0c35f2